src/snowflake/snowpark/_internal/compiler/plan_compiler.py

```diff
@@ -48,13 +57,24 @@ def should_apply_optimizations(self) -> bool:
         ) and current_session.cte_optimization_enabled
 
     def compile(self) -> Dict[PlanQueryType, List[Query]]:
```

Comment:
this method will not be triggered by any existing code path, right? (outside of query optimization)

Reply:
No, this method is already triggered today for execution_queries in general, however, under the case when no optimization is enabled, it is doing the same as today.